### PR TITLE
OpenSslSession#initPeerCerts creates too long almost empty arrays.

### PR DESCRIPTION
**Pull request by ichaki5748**
Motivation:

https://github.com/netty/netty/issues/5945

Modifications:

Refactored initialization of arrays. Fixed arrays length

Result:

Cert arrays have proper length. Testing added

### CODE REVIEW

File: handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java [hunk 1]
```diff
@@ -1747,54 +1749,56 @@ void handshakeFinished() throws SSLException {
...
-                }
+        private void initCerts(byte[][] chain, int startPos) {
+            for (int i = 0; i < chain.length; i++) {
+                int certPos = startPos + i;
+                peerCerts[certPos] = new OpenSslX509Certificate(chain[i]);
+                x509PeerCerts[certPos] = new OpenSslJavaxX509Certificate(chain[i]);
             }
         }
 
+        private boolean isEmpty(Object[] arr) {
```
Comment (netkins) [inline handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java]: MINOR Consider using varargs for methods or constructors which take an array the last parameter. [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUseVarargs)

Comment (normanmaurer) [inline handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java]: @ichaki5748 can you make this method static

File: handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java [hunk 2]
```diff
@@ -1747,54 +1749,56 @@ void handshakeFinished() throws SSLException {
...
+            return arr == null || arr.length == 0;
+        }
+
+        private boolean isEmpty(byte[] cert) {
```
Comment (normanmaurer) [inline handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java]: please make static

Comment (ichaki5748) [inline handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java]: sure

**Comment by normanmaurer** [comment]
@ichaki5748 please do the requested changes and squash. I will cherry-pick after it.

File: handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java [hunk 3]
```diff
@@ -1308,6 +1310,14 @@ private void checkEngineClosed(SSLException cause) throws SSLException {
         return pendingStatus > 0 ? NEED_WRAP : NEED_UNWRAP;
     }
 
+    private static boolean isEmpty(Object[] arr) {
```
Comment (netkins) [inline handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java]: MINOR Consider using varargs for methods or constructors which take an array the last parameter. [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUseVarargs)

**Comment by normanmaurer** [comment]
Cherry-picked into 4.1 (4ee361e3024b33ac3a160e8124ee7f0e5a1f4b42) and 4.0 (a7662db47040c5a923d38fda421b4800cf983403)

**Pull request closed without merging**